flight/mcu/lib/HistoryBuffer/HistoryBuffer.h

```diff
@@ -0,0 +1,144 @@
+#ifndef TARS_SOFTWARE_HISTORY_BUFFER
+#define TARS_SOFTWARE_HISTORY_BUFFER
+
+#include <cstdint>
+#include <cstring>
+
+#ifdef COMPILE_LOCAL
+#include <mutex>
+#endif
+
+#include <ChRt.h>
+
+
+/*
+History Fifo Buffer
+*/
+template <size_t size>
+class HistoryBuffer{
+    private:
+        float arr[size];
+        systime_t timestampArr[size];
+
+        int frontIndex;
+        int endIndex;
+    
+    public:
+        HistoryBuffer(){
+            frontIndex = 0;
+            endIndex = size - 1;
+        }
+
+        void push(float val, float timestamp){
+            //sets the value at the end of the queue to be equal to the desired push value
+            arr[endIndex] = val;
+            timestampArr[endIndex] = timestamp;
+
+            //since our newest value is at the end index, it becomes our start index
+            frontIndex = endIndex;
+
+            //sets the new end index to the next oldest value which will be one less than the previous end index
+            //if its currently 0, it needs to wrap around to the back
+            if(endIndex == 0){
+                endIndex = size-1;
+            }
+            else{
+                endIndex--;
+            }
```

Comment:
All of this logic seems very wrong. If this logic is to be believed, you don't need a `frontIndex` at all, you can just add 1 to `endIndex` (and wrap if necessary) to get `frontIndex`, always.

Reply:
I see your point about front being unnecessary, I felt that it was better for clarity. What about the logic seems wrong?